backend/src/main/java/com/htecgroup/skynest/service/impl/UserServiceImpl.java

```diff
@@ -198,11 +200,20 @@ public void demoteUser(
   }
 
   @Override
-  public void addCompanyForUser(
-      @Valid @UserNotInACompany UUID uuid, @Valid @CurrentUserIsInCompany UUID companyId) {
-    CompanyDto companyDto = companyService.findById(companyId);
+  public void addCompanyForUser(@Valid @UserNotInACompany @UserNotAdmin UUID uuid) {
+    CompanyEntity companyEntity =
+        currentUserService
+            .getCompanyEntityFromLoggedUser()
+            .orElseThrow(UserNotInAnyCompanyException::new);
     UserDto userDto = findUserById(uuid);
```

Comment:
Maybe use the regular find (repo method) here, so we don't have to map from Dto to UserEntity.

Reply:
But this is what's done everywhere, that's gonna be left for one big refactor i think.

Reply:
Yep, I approve